File: nim/test_c2nim/calc.nim
 {.deadCodeElim: on.}
when defined(windows):
  const
    calcdll* = "calc.dll"
elif defined(macosx):
  const
    calcdll* = "libcalc.dylib"
else:
  const
    calcdll* = "libcalc.so"
const
  MY_CONSTANT* = 10

proc add*(x: cint; y: cint): cint {.cdecl, importc: "add", dynlib: calcdll.}
proc sumArray*(length: cint; array: ptr cint): cint {.cdecl, importc: "sum_array",
    dynlib: calcdll.}

File: zig/translate-c/calc.c

int add(int x, int y)
{
	return x + y;
}


int sum_array(int length, int *array)
{
	int total = 0;

	for (int i = 0; i < length; i++) {
		total += array[i];
	}

	return total;
}